### PR TITLE
Migrate to php-parser v2.x

### PR DESCRIPTION
**Pull request by emmetog**
This PR is a resurrection of https://github.com/vim-php/phpctags/pull/71.

It fixes this error:

```
PHP Fatal error:  Uncaught Error: Class 'PHPParser_Parser' not found in ~/.composer/vendor/techlivezheng/phpctags/PHPCtags.class.php:31
Stack trace:
#0 ~/.composer/vendor/techlivezheng/phpctags/bootstrap.php(236): PHPCtags->__construct(Array)
#1 ~/.composer/vendor/techlivezheng/phpctags/bin/phpctags(4): include('/home/emmet/.co...')
#2 {main}
  thrown in ~/.composer/vendor/techlivezheng/phpctags/PHPCtags.class.php on line 31
```

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -15,8 +12,3 @@ before_script:
 
 script:
     - vendor/bin/phpunit
-
-matrix:
-  fast_finish: true
-  allow_failures:
-    - php: 7
```
Comment (emmetog) [inline .travis.yml]: This means that the tests must pass for php7 too

File: .travis.yml [hunk 2]
```diff
@@ -1,9 +1,6 @@
 language: php
 
 php:
-    - 5.3
-    - 5.4
-    - 5.5
```
Comment (emmetog) [inline .travis.yml]: I've removed versions which are no longer supported: http://php.net/supported-versions.php

Comment (Rican7) [inline .travis.yml]: While I understand your intent here, you're removing versions that the `composer.json` file says the library is compatible with: https://github.com/vim-php/phpctags/pull/73/files#diff-b5d0ee8c97c7abd7e3fa29b9a27d1780R7.

Comment (emmetog) [inline .travis.yml]: Good catch, I've updated the `composer.json` to only support PHP >= 5.6

Comment (MelleB) [inline .travis.yml]: Minor: This means the README should be updated as well.

Comment (emmetog) [inline .travis.yml]: @MelleB done

**Comment by jostillmanns** [comment]
shouldn't we directly go for version 3.x?

**Comment by emmetog** [comment]
@jostillmanns Personally I prefer many little improvements because they are easier to test and carry less risk that something can break so I vote to update to 3.x in a separate PR. This PR is already a win, we're updating to 2.x.

If the maintainers feel strongly for going straight to 3.x though I'll update this PR.

**Comment by lvht** [comment]
I made a new one #78

**Comment by InFog** [comment]
#78 was already merged updating php-parser to 3.x. There is no need to keep this one open.

Thank you, @emmetog for the effort and thank you @lvht for #78.

**Pull request closed without merging**